controller/src/beerocks/master/tasks/association_handling_task.cpp

```diff
@@ -70,18 +71,39 @@ void association_handling_task::work()
     case START_RSSI_MONITORING: {
         TASK_LOG(DEBUG) << "START_RSSI_MONITORING";
         /*
-            * request constant RSSI monitoring for the new client
-            */
+         * request constant RSSI monitoring for the new client
+         */
+
+        bool send_start_monitoring_request = true;
+        std::string new_hostap_mac;
+        std::string ipv4;
+
         if (database.is_node_wireless(sta_mac)) {
-            std::string new_hostap_mac = database.get_node_parent(sta_mac);
+            new_hostap_mac = database.get_node_parent(sta_mac);
             if (new_hostap_mac != original_parent_mac ||
                 database.get_node_state(sta_mac) != beerocks::STATE_CONNECTED) {
                 TASK_LOG(DEBUG) << "sta " << sta_mac << " is no longer connected to "
                                 << original_parent_mac << " finishing task";
                 finish();
                 return;
             }
-            std::string ipv4 = database.get_node_ipv4(sta_mac);
+
```

Comment:
I think that it will be much more readable like this:

```
bool send_start_monitoring_request = false;
...
if (!ipv4.empty()) {
    send_start_monitoring_request = true;
}

if (send_start_monitoring_request) {
    ...
} else {
    // handle timeouts
}
```

Reply:
I disagree with that one since IMO, we should first handle errors and then the rest of the code. This makes the code more readable.